### DISCUSSION TITLE
[ZK Stack] - How to deploy zkstack in docker?

### DISCUSSION THREAD

**Discussion by SuccinctPaul** [zkSync-Community-Hub/zksync-developers#627] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.78.0-nightly

### Operating System

- [x] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

How to deploy zkstack in docker with customer token?
And what's the difference between server-v2 and local-node dockerfile?

### Expected Behavior

Generate a docker-compose file to start zkstack like  https://github.com/matter-labs/local-setup.git did.

### Current Behavior

Non idea.

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment, marked as answer]
To deploy ZK Stack in Docker, you can follow these general steps:

1. **Clone the ZKsync Era repository**:
   ```bash
   git clone https://github.com/matter-labs/zksync-era.git
   cd zksync-era
   ```

2. **Set up Docker**:
   Ensure Docker and Docker Compose are installed on your machine. If not, you can install them from [Docker's official site](https://docs.docker.com/get-docker/).

3. **Generate Docker Compose File**:
   Use the `zk stack docker-setup` command to generate docker images for your server and prover. This will guide you through naming and tagging your image. After building it, a docker-compose file will be available for running the images in your desired environment.

4. **Run Docker Compose**:
   Navigate to the directory containing your docker-compose file and start your ZK Stack:
   ```bash
   docker-compose up
   ```

Regarding the difference between `server-v2` and `local-node` Dockerfiles:

- **server-v2**: Typically refers to a newer version of the server setup, possibly with updated configurations or enhancements specific to the server's operation in a production or test environment.

- **local-node**: Designed for local testing and development. This setup is usually configured to simulate the environment with less complexity and more debugging features.

For a more detailed guide and custom configurations, you might want to refer to the [ZKsync Node Quick Start Guide](https://docs.zksync.io/zksync-node/quickstart) and the [Docker L1 - L2 Nodes documentation](https://docs.zksync.io/build/test-and-debug/dockerized-l1-l2-nodes). These resources provide comprehensive instructions on setting up ZKsync nodes using Docker.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Reply by Amyevesky** [reply]
I tried making a docker image, but this command is not working in the ecosystem, "zk stack docker-setup". I am trying to run this command through zkstack cli but it returens command not found, can you please tell me what's the issue ? @SuccinctPaul

**Comment by bxpana** [comment]
Hey @SuccinctPaul it seems like you've asked in a few different posts about the custom base token. Are they all related or are each different? https://github.com/zkSync-Community-Hub/zksync-developers/discussions/622#top and https://github.com/zkSync-Community-Hub/zksync-developers/discussions/626#top

**Reply by SuccinctPaul** [reply]
@bxpana Hi，they are related.  
#626 is a sub-question of #622

**Reply by bxpana** [reply]
So first you're asking about how to deploy and use the new base token in https://github.com/zkSync-Community-Hub/zksync-developers/discussions/622 and then how to bridge it from L1 to L2 (locally) in https://github.com/zkSync-Community-Hub/zksync-developers/discussions/626 ?

**Reply by SuccinctPaul** [reply]
> So first you're asking about how to deploy and use the new base token in #622 and then how to bridge it from L1 to L2 (locally) in #626 ?

Yep

**Reply by bxpana** [reply]
Seems like the response from the bot should have helped answer the questions with docker, but regarding the base token we're working on the docs and should have them out within the next week. Will keep you posted.

**Comment by Amyevesky** [comment]
I tried making a docker image, but this command is not working in the ecosystem, "zk stack docker-setup". I am trying to run this command through zkstack cli but it returens command not found, can you please tell me what's the issue ?
